### PR TITLE
Configure Node server to forward requests to Pulsar

### PR DESCRIPTION
**Pull request by mfortman11**
How to set it up (this is will be added to the README when the other tasks are also completed)

1. Run pulsar standalone `docker run -it -p 6650:6650  -p 8080:8080 --mount source=pulsardata,target=/pulsar/data --mount source=pulsarconf,target=/pulsar/conf apachepulsar/pulsar:2.8.3 bin/pulsar standalone`
2.  Optional: If rerunning pulsar standalone on a different port that than `8080` above go to `server/config.js` and update the `PULSAR_URL`
3. Start the server (`cd server` and `npm run start`)
4. Run the console (`cd dashboard` and `npm run serve`)

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -42,6 +44,38 @@ process.env['NODE_ENV']='production'
 // place holder for the data
 const users = [];
 
+app.use('/ws/', createProxyMiddleware({
+  target: cfg.serverConfig.WEBSOCKET_URL,
+  'option.ws': true,
+}));
+
+app.use('/api/v1/standalone/functions', createProxyMiddleware({
```
Comment (cdbartholomew) [inline server/server.js:52]: The name of the cluster, ie `standalone`, needs to be passed in by configuration. From what I can tell, this code will only work with a development Pulsar instance. For example, in Kubernetes, the default cluster name is `pulsar` and the endpoints need to be in the form `/api/v1/pulsar/functions` and the rewrite needs to strip out `pulsar` and change the endpoint to `/admin/v3` (in the case of functions).

Comment (cdbartholomew) [reply server/server.js:52]: There are 2 places in the `dashboard/public/config.js` file to configure the cluster name:
    "home_cluster": "standalone",
    "cluster_list": ["standalone"],
There is code to support multiple clusters, but it is not fully exposed. So this is just a list of 1 cluster. The "home_cluster" is the one it starts at on the list.

**Review by cdbartholomew — CHANGES_REQUESTED** [review]

File: server/server.js [hunk 2]
```diff
@@ -42,6 +44,38 @@ process.env['NODE_ENV']='production'
...
+
+app.use('/api/v1/standalone/sources', createProxyMiddleware({
+  target: cfg.serverConfig.PULSAR_URL,
+  pathRewrite: {'^/api/v1/standalone/sources/': '/admin/v3/sources/'},
+  'option.headers': {Accept: 'application/json/'},
+  'option.toProxy': true
+}));
+
+app.use('/api/v1/standalone', createProxyMiddleware({
+  target: 'http://localhost:8080',
```
Comment (cdbartholomew) [inline server/server.js:74]: I think this should be set to `cfg.serverConfig.PULSAR_URL`, not localhost.

File: server/config.js [hunk 3]
```diff
@@ -38,6 +38,8 @@ let dashboardConfig = {
 
 let serverConfig = {
     'PORT': '6454',
+    'PULSAR_URL': 'http://localhost:8080',
+    'WEBSOCKET_URL': 'https://websocket.example.com:8500',
     'LOG_LEVEL': 'info',
```
Comment (cdbartholomew) [inline server/config.js:43]: To completely remove nginx, the server will also have to support TLS. Previously, nginx was providing that ability. There are default certs used by nginx and the vue dev server that can be used to test TLS configuration.

Comment (cdbartholomew) [reply server/config.js:43]: Actually, it looks like the server might already support TLS.

**Comment by mfortman11** [comment]
All config options are now in `dashboard/public/config.js`

**Pull request merged**